shell/platform/darwin/macos/README.md

```diff
@@ -0,0 +1,44 @@
+# Flutter macOS Embedder
+
+This directory contains files specific to the Flutter macOS embedder. The
+embedder is built as a macOS framework that is linked into the target
+application. The framework is composed both of macOS-specific code and code
+shared with the iOS embedder. These can be found in:
+```
+flutter/shell/platform/darwin/common/framework
+flutter/shell/platform/darwin/macos/framework
+```
+
+Additionally, the framework relies on utility code shared across all embedders,
+which is found in:
+```
+flutter/shell/platform/common
+```
+
+## Building
+
+Building all artifacts required for macOS occurs in two steps:
+* Host binary build: Builds tooling used by the Flutter tool to to build Flutter
+  applications targeting macOS.
+* macOS Target build: Builds the framework that implements the macOS Flutter
+  embedder and exposes public API such as `FlutterViewController` used by
+  Flutter applications.
+
+To build the macOS embedder from the `src/flutter` directory, run the following
+commands:
+```sh
+# Perform the host build.
+./tools/gn --unopt                           # Non-Googlers should add --no-goma
```

Comment:
I think it would be better to use `./tools/gn --unopt --no-goma` and then have the comment say `# Goolers should remove --no-goma`

Then external people won't have to read the comments, and Googlers get rewarded if they do (but only slightly slowed down if they don't). Same below.

Reply:
Done.